x-pack/plugins/cloud_security_posture/public/pages/configurations/findings_flyout/rule_tab.tsx

```diff
@@ -5,48 +5,50 @@
  * 2.0.
  */
 
-import { EuiBadge, EuiDescriptionList, EuiLink, EuiToolTip } from '@elastic/eui';
+import { EuiBadge, EuiDescriptionList } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import React from 'react';
 import { FormattedMessage } from '@kbn/i18n-react';
 import { CspFinding } from '../../../../common/schemas/csp_finding';
 import { RulesDetectionRuleCounter } from '../../rules/rules_detection_rule_counter';
-import { CisKubernetesIcons, CspFlyoutMarkdown } from './findings_flyout';
+import {
+  CisKubernetesIcons,
+  CspFlyoutMarkdown,
+  EMPTY_VALUE,
+  RuleNameLink,
+} from './findings_flyout';
 
 export const getRuleList = (
-  rule: CspFinding['rule'],
+  rule?: CspFinding['rule'],
   ruleState = 'unmuted',
   ruleFlyoutLink?: string
 ) => [
   {
     title: i18n.translate('xpack.csp.findings.findingsFlyout.ruleTab.nameTitle', {
       defaultMessage: 'Name',
     }),
-    description: ruleFlyoutLink ? (
-      <EuiToolTip
-        position="top"
-        content={i18n.translate('xpack.csp.findings.findingsFlyout.ruleTab.nameTooltip', {
-          defaultMessage: 'Manage Rule',
-        })}
-      >
-        <EuiLink href={ruleFlyoutLink}>{rule.name}</EuiLink>
-      </EuiToolTip>
+    description: rule?.name ? (
+      <RuleNameLink ruleFlyoutLink={ruleFlyoutLink} ruleName={rule.name} />
     ) : (
-      rule.name
+      EMPTY_VALUE
     ),
   },
   {
     title: i18n.translate('xpack.csp.findings.findingsFlyout.ruleTab.descriptionTitle', {
       defaultMessage: 'Description',
     }),
-    description: <CspFlyoutMarkdown>{rule.description}</CspFlyoutMarkdown>,
+    description: rule?.description ? (
+      <CspFlyoutMarkdown>{rule.description}</CspFlyoutMarkdown>
+    ) : (
+      EMPTY_VALUE
+    ),
   },
   {
     title: i18n.translate('xpack.csp.findings.findingsFlyout.ruleTab.AlertsTitle', {
       defaultMessage: 'Alerts',
     }),
     description:
-      ruleState === 'unmuted' ? (
+      ruleState === 'unmuted' && rule?.benchmark?.name ? (
```

Comment:
why this check? If it's `RulesDetectionRuleCounter` relies on this field somewhere deep inside, I think we should update the function, not add this check here as it's very hard to see the connection between missing benchmark name and not showing detection rule counter

Reply:
Why? the type for `RulesDetectionRuleCounter` says it expect to get the benchmark object without missing fields. if you pass an optional value it should throw a type error. Meaning we should check the fields exists before passing them to the component. The component itself does not necessarily have to provide a fallback render, it just have to function with what it expects.

Reply:
I guess it's again more of a subjective matter. `ruleState === 'unmuted'` is purely a business logic and `rule?.benchmark?.name` is an internal requirement of `<RulesDetectionRuleCounter benchmarkRule={rule} />` and we are mixing them here in one check. TS definitely helps with knowing about the internal requirement, it's just looking at the code without checking the types that made me stop and think, why it is here.